tests/cancun/eip6780_selfdestruct/test_selfdestruct_revert.py

```diff
@@ -0,0 +1,311 @@
+            case 0 /* no-op used for transfering value to this contract */ {{
+                return(0, 0)
+            }}
+            case 1 {{
+                selfdestruct({selfdestruct_recipient_address})
+            }}
+            default /* unsupported operation */ {{
+                stop()
+            }}
+        }}
+        """
+    )
+
+
+@pytest.fixture
+def selfdestruct_with_transfer_contract_initcode(
+    selfdestruct_with_transfer_contract_code: SupportsBytes,
+) -> SupportsBytes:
+    """Initcode for selfdestruct_with_transfer_contract_code"""
+    return Initcode(deploy_code=selfdestruct_with_transfer_contract_code)
+
+
+@pytest.fixture
+def selfdestruct_with_transfer_initcode_copy_from_address() -> str:
+    """Address of a pre-existing contract we use to simply copy initcode from."""
+    return to_address(0xABCD)
+
+
+@pytest.fixture
+def pre(
+    recursive_revert_contract_address: str,
+    recursive_revert_contract_code: SupportsBytes,
+    selfdestruct_with_transfer_initcode_copy_from_address: str,
+    selfdestruct_with_transfer_contract_initcode: SupportsBytes,
+    selfdestruct_with_transfer_contract_address: str,
+    yul: YulCompiler,
+) -> Dict[str, Account]:
+    """Prestate for test_selfdestruct_not_created_in_same_tx_with_revert"""
+    return {
+        TestAddress: Account(balance=100_000_000_000_000_000_000),
+        selfdestruct_with_transfer_initcode_copy_from_address: Account(
+            code=selfdestruct_with_transfer_contract_initcode
+        ),
+        recursive_revert_contract_address: Account(code=recursive_revert_contract_code, balance=2),
+    }
+
+
+@pytest.mark.valid_from("Shanghai")
+def test_selfdestruct_created_in_same_tx_with_revert(  # noqa SC200
+    state_test: StateTestFiller,
+    env: Environment,
+    pre: Dict[str, Account],
+    entry_code_address: str,
+    selfdestruct_with_transfer_contract_code: SupportsBytes,
+    selfdestruct_with_transfer_contract_initcode: SupportsBytes,
+    selfdestruct_with_transfer_contract_address: str,
+    selfdestruct_recipient_address: str,
+    selfdestruct_with_transfer_initcode_copy_from_address: str,
+    recursive_revert_contract_address: str,
+):
+    """
+    Given:
+        contract A which has methods to receive balance and selfdestruct, and was created in current tx
+    test the following call sequence in a tx:
+         transfer value to A and call A.selfdestruct.
+         recurse into a new call from transfers value to A, calls A.selfdestruct, and reverts.
+    """  # noqa: E501
+    entry_code = Op.EXTCODECOPY(
+        Op.PUSH20(selfdestruct_with_transfer_initcode_copy_from_address),
+        # TODO: should detect the size of the address automatically ^
+        0,
+        0,
+        len(bytes(selfdestruct_with_transfer_contract_initcode)),
+    )
+
+    entry_code += Op.CREATE(
+        0, 0, len(bytes(selfdestruct_with_transfer_contract_initcode))  # Value  # Offset
+    )
+
+    entry_code += Op.POP()
```

Comment:
```suggestion
entry_code += Op.SSTORE(
        0,
        Op.CREATE(
            0, 0, len(bytes(selfdestruct_with_transfer_contract_initcode))  # Value  # Offset
        ),
    )
```
We could save the address and verify it for good measure. I think this just requires adding a return at the end of `entry_code` to be able to verify `storage={0: selfdestruct_with_transfer_contract_address}`.